src/main/java/org/ibp/api/brapi/v2/program/ProgramResourceBrapi.java

```diff
@@ -0,0 +1,88 @@
+package org.ibp.api.brapi.v2.program;
+
+import com.fasterxml.jackson.annotation.JsonView;
+import com.google.common.collect.ImmutableMap;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.apache.commons.lang3.StringUtils;
+import org.generationcp.middleware.service.api.program.ProgramSearchRequest;
+import org.generationcp.middleware.service.api.BrapiView;
+import org.generationcp.middleware.service.api.program.ProgramDetailsDto;
+import org.ibp.api.Util;
+import org.ibp.api.brapi.v1.common.*;
+import org.ibp.api.brapi.v1.program.ProgramEntityResponse;
+import org.ibp.api.domain.common.PagedResult;
+import org.ibp.api.exception.ApiRuntimeException;
+import org.ibp.api.java.program.ProgramService;
+import org.ibp.api.rest.common.PaginatedSearch;
+import org.ibp.api.rest.common.SearchSpec;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@Api(value = "BrAPI V2 Program Services")
+@Controller(value = "ProgramResourceBrapiV2")
+public class ProgramResourceBrapi {
+
+    @Autowired
+    private ProgramService programService;
+
+    @ApiOperation(value = "Get filtered list of breeding Programs", notes = "Get a filtered list of breeding Programs. This list can be filtered by common crop name to narrow results to a specific crop.")
+    @RequestMapping(value = "/{commonCropName}/brapi/v2/programs", method = RequestMethod.GET)
+    @ResponseBody
+    @JsonView(BrapiView.BrapiV2.class)
+    public ResponseEntity<EntityListResponse<ProgramDetailsDto>> listPrograms(
+            @ApiParam(value = "Filter by the common crop name. Exact match.")
+            @PathVariable final String commonCropName,
```

Comment:
As discussed,
1.  commonCropName is a filter not URL path variable
2. retrieve list of crops accessible to logged in user and only return programs for those crop/s. Perform validation that commonCropName paramter, if existing, is part of that list of crops